datagateway_api/src/search_api/helpers.py

```diff
@@ -74,8 +78,69 @@ def create_error_message(e):
     return wrapper_error_handling
 
 
+def get_score(entities, query):
+    """
+    Gets the score on the given entities based in the query parameter
+    that is the term to be found
+    :param entities: List of entities that have been retrieved from one ICAT query.
+    :type entities: :class:`list`
+    :param query: String with the term to be searched by
+    :type query: :class:`str`
+    """
+    try:
+        data = {
+            "query": query,
+            "group": Config.config.search_api.scoring_group,
+            "limit": Config.config.search_api.scoring_limit,
+            # With itemIds, scoring server returns a 400 error. No idea why.
+            # "itemIds": list(map(lambda entity: (entity["pid"]), entities)),  #
+        }
+        response = requests.post(
+            Config.config.search_api.scoring_server,
+            json=data,
+            timeout=5,  # Could this be a configuration parameter?
+        )
+        if response.status_code < 400:
+            return response.json()["scores"]
+        else:
+            raise ScoringAPIError(
+                Exception(f"Score API returned {response.status_code}"),
+            )
+    except ValueError as e:
+        log.error("Response is not a valid json")
+        raise e
+    except ConnectionError as e:
+        log.error("ConnectionError to %s ", Config.config.search_api.scoring_server)
+        raise e
```

Comment:
This exception is not caught at a higher level as part of the `search_api_error_handling` decorator method, meaning that the whole stack trace is returned to the consumer. We don't want to do this as it is considered bad security practice. It might be best if we raise `ScoringAPIError` here and also in the places where the `ValueError` and `Exception` are caught.